File: 算法/代码随想录算法PDF/00调试源代码/150.逆波兰表达式求值.cpp
/*
 * @lc app=leetcode.cn id=150 lang=cpp
 *
 * [150] 逆波兰表达式求值
 */

// @lc code=start
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for (string s : tokens) {
            if (s == "+" || s == "-" || s == "*" || s == "/") {
                int num1 = st.top();
                st.pop();
                int num2 = st.top();
                st.pop();
                if (s == "+") st.push(num2 + num1);
                if (s == "-") st.push(num2 - num1);
                if (s == "*") st.push(num2 * num1);
                if (s == "/") st.push(num2 / num1);
            } else {
                st.push(stoll(s));
            }
        }
        return st.top();
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/150.逆波兰表达式求值.py
#
# @lc app=leetcode.cn id=150 lang=python3
#
# [150] 逆波兰表达式求值
#

# @lc code=start
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for s in tokens:
            if s in {"+", "-", "*", "/"}:
                num1 = st.pop()
                num2 = st.pop()
                st.append(int(eval(f'{num2} {s} {num1}')))
            else:
                st.append(int(s))
        return st.pop()
# @lc code=end

